### PR TITLE
[MAINTENANCE] namespace snippets

### PR DESCRIPTION
**Pull request by tyler-hoffman**
Updates our snippet checking logic to be namespaced by version directory (`docs/`, `versioned_docs/<VERSION>/`, etc. This allows us to remove the last file processing in `prepare_prior_versions.py`

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/docusaurus/scripts/remark-named-snippets/snippet.js [hunk 1]
```diff
@@ -191,27 +189,33 @@ function processVerbose() {
...
-  const out = {}
-  for (const snippet of snippets) {
-    // If no explicit args are provided, default to all snippets
-    // Else, ensure that the snippet's source file was requested by the user
-    const file = snippet.file
-    if (targetFiles.length > 0 && !(targetFiles.includes(file))) {
-      continue
+  const dirs = getDirs()
+  for (let dir of dirs) {
+    const snippets = parseSourceDirectories(dir)
```
Comment (cdkini) [inline docs/docusaurus/scripts/remark-named-snippets/snippet.js]: Should this now be `parseSourceDirectory`?

File: docs/docusaurus/scripts/remark-named-snippets/snippet.js [hunk 2]
```diff
@@ -191,27 +189,33 @@ function processVerbose() {
...
+      delete snippet.file // Remove duplicate filename to clean up stdout
+      out[file].push(snippet)
     }
-    if (!(file in out)) {
-      out[file] = []
+    if (verbose) {
+      // printing objects one dir at a time so the output is more readable
+      console.log("Snippets found in " + dir)
+      console.log("---------------------------")
+      console.log(out)
```
Comment (cdkini) [inline docs/docusaurus/scripts/remark-named-snippets/snippet.js:217]: Will this print legibly to the console or do we need to iterate through?

**Review by cdkini — APPROVED** [review]

File: docs/docusaurus/scripts/remark-named-snippets/snippet.js (hunk 2 above)
Comment (tyler-hoffman) [reply docs/docusaurus/scripts/remark-named-snippets/snippet.js:217]: It prints legibly. I tried putting `out` for each directory into a larger dictionary and started seeing `[Object object]` (or something like that) for leaf nodes.

File: docs/docusaurus/scripts/remark-named-snippets/snippet.js (hunk 1 above)
Comment (tyler-hoffman) [reply docs/docusaurus/scripts/remark-named-snippets/snippet.js]: good catch. Updated!

**Pull request merged**